extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -951,9 +946,14 @@ export const popupMessageListener = (request: Request) => {
     };
   };
 
-  const loadSettings = () => {
-    const dataSharingValue = localStorage.getItem(DATA_SHARING_ID) || "true";
-    const isDataSharingAllowed = JSON.parse(dataSharingValue);
+  const loadSettings = async () => {
+    await migrateLocalStorageToBrowserStorage();
```

Comment:
`loadSettings` runs very often (every time you open the extension) so this seems like a good place to keep the 2 data stores in sync